virtualization.h

```diff
@@ -21,7 +21,7 @@ bool shouldAcceptNewConnectionHandler(void *listener, void *connection, void *so
 
 /* VZVirtioSocketListener */
 @interface VZVirtioSocketListenerDelegateImpl : NSObject <VZVirtioSocketListenerDelegate>
-- (BOOL)listener:(VZVirtioSocketListener *)listener shouldAcceptNewConnection:(VZVirtioSocketConnection *)connection fromSocketDevice:(VZVirtioSocketDevice *)socketDevice;
+- (BOOL)listener:(void *)listener shouldAcceptNewConnection:(void *)connection fromSocketDevice:(void *)socketDevice;
```

Comment:
This is indirectly related as this silences some warnings when building on older macOS versions. I've split the changes to a separate commit with an explanation.

Reply:
I'm not sure what you mean. This library is supported only above macOS v11.0. So I think this is no problem.

Reply:
https://github.com/Code-Hex/vz/issues/41 asks for macOS v10 'support'. There will be no virtualization on these platforms, but errors/exceptions will be returned. Returning errors/exceptions need to be done in `Code-Hex/vz`. To add these, I use a lot compiler warnings:
```
     # github.com/Code-Hex/vz/v2
    In file included from _cgo_export.c:4:
    In file included from socket.go:6:
    ../../virtualization.h:24:19: warning: 'VZVirtioSocketListener' is only available on macOS 11.0 or newer [-Wunguarded-availability-new]
    /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Virtualization.framework/Headers/VZVirtioSocketListener.h:25:12: note: 'VZVirtioSocketListener' has been marked as being introduced in macOS 11.0 here, but the deployment target is macOS 10.13.0
    ../../virtualization.h:24:1: note: annotate 'listener:shouldAcceptNewConnection:fromSocketDevice:' with an availability attribute to silence this warning
    ../../virtualization.h:24:80: warning: 'VZVirtioSocketConnection' is only available on macOS 11.0 or newer [-Wunguarded-availability-new]
    /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Virtualization.framework/Headers/VZVirtioSocketConnection.h:18:12: note: 'VZVirtioSocketConnection' has been marked as being introduced in macOS 11.0 here, but the deployment target is macOS 10.13.0
    ../../virtualization.h:24:1: note: annotate 'listener:shouldAcceptNewConnection:fromSocketDevice:' with an availability attribute to silence this warning
    ../../virtualization.h:24:136: warning: 'VZVirtioSocketDevice' is only available on macOS 11.0 or newer [-Wunguarded-availability-new]
    /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Virtualization.framework/Headers/VZVirtioSocketDevice.h:25:12: note: 'VZVirtioSocketDevice' has been marked as being introduced in macOS 11.0 here, but the deployment target is macOS 10.13.0
    ../../virtualization.h:24:1: note: annotate 'listener:shouldAcceptNewConnection:fromSocketDevice:' with an availability attribute to silence this warning
```

If there are warnings, there is some code without `@available`. If there are pages and pages of warnings, I will not notice new code missing `@available`. This is why I prefer to remove all the warnings I can.

Reply:
My opinion. https://github.com/Code-Hex/vz/issues/41#issuecomment-1265260233

I don't support the compilation on macOS v10. because It depends on xcode sdk.
So you don't need to consider for v10.

Reply:
These warnings happen with latest macOS/xcode when you build a binary for macOS 10 or newer:
```zsh
% sw_vers 
ProductName:	macOS
ProductVersion:	12.6
BuildVersion:	21G115

% CGO_CFLAGS=-mmacosx-version-min=10.13  make -C example/linux
go build -o virtualization .
# github.com/Code-Hex/vz/v2
In file included from virtualization.m:8:
../../virtualization_view.h:26:41: warning: 'VZVirtualMachine' is only available on macOS 11.0 or newer [-Wunguarded-availability-new]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Virtualization.framework/Headers/VZVirtualMachine.h:67:12: note: 'VZVirtualMachine' has been marked as being introduced in macOS 11.0 here, but the deployment target is macOS 10.13.0
../../virtualization_view.h:26:1: note: annotate 'initWithVirtualMachine:windowWidth:windowHeight:' with an availability attribute to silence this warning
virtualization.m:1249:37: warning: 'VZVirtualMachine' is only available on macOS 11.0 or newer [-Wunguarded-availability-new]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Virtualization.framework/Headers/VZVirtualMachine.h:67:12: note: 'VZVirtualMachine' has been marked as being introduced in macOS 11.0 here, but the deployment target is macOS 10.13.0
virtualization.m:1249:37: note: enclose 'VZVirtualMachine' in an @available check to silence this warning
# github.com/Code-Hex/vz/v2
```

They are not related to building on macOS 10. I agree we don't need to support *building* on macOS.

Reply:
Please revert this~

Reply:
@cfergeau Ping